src/transformers/trainer.py

```diff
@@ -1956,6 +1978,17 @@ def _inner_training_loop(
         # Wait for the checkpoint to be uploaded.
         self._finish_current_push()
 
+        # After training we make sure to retrieve back the original forward pass method
+        # for the embedding layer by removing the forward post hook.
+        if self.neftune_noise_alpha is not None:
+            if is_peft_available() and isinstance(self.model, PeftModel):
+                embeddings = unwrap_model(self.model.base_model).get_input_embeddings()
+            else:
+                embeddings = unwrap_model(self.model).get_input_embeddings()
+
+            self.neftune_hook_handle.remove()
+            del embeddings.neftune_noise_alpha
```

Comment:
Let's make this into an equivalent method `_deacivate_neftune`

Reply:
Done in e55ab8b